docs/howto/code/custom-action-updatelogfile/updatelogfile_action_charm.py

```diff
@@ -0,0 +1,57 @@
+#!/usr/bin/env python3
+# Copyright 2025 Ubuntu
+# See LICENSE file for licensing details.
+
+"""Flask Charm entrypoint."""
+
+import logging
+import typing
+
+import ops
+
+import paas_charm.flask
+
+import requests
+
+logger = logging.getLogger(__name__)
+
+
+class FlaskHelloWorldCharm(paas_charm.flask.Charm):
+    """Flask Charm service."""
+
+    def __init__(self, *args: typing.Any) -> None:
+        """Initialize the instance.
+
+        Args:
+            args: passthrough to CharmBase.
+        """
+        super().__init__(*args)
+        self.framework.observe(self.on.updatelogfile_action,
+                               self._on_updatelogfile_action)
+
+    def _on_updatelogfile_action(self, event: ops.ActionEvent) -> None:
+        """Handle the updatelogfile action.
+
+        Args:
+            event: the action event object.
+            logfile: the output logfile in the container
+        """
+        if not self.is_ready():
+            event.fail("flask-app container is not ready")
+        try:
+            response = requests.get(
+                    f"http://127.0.0.1:{self._workload_config.port}", timeout=5
+                    )
+            response.raise_for_status()
+            self._container.push(event.params["logfile"], response.text)
+            output = response.text
+            output += " written to file " + event.params["logfile"]
+            event.set_results({"result": output})
+        except ops.pebble.ExecError as e:
+            event.fail(str(e.stderr))
+        except requests.exceptions.RequestException as e:
+            # if it failed with http bad status code or the connection failed
+            event.fail(str(e.stderr))
```

Comment:
`RequestException` won't have a stdout/stderr attribute. But this should work (I just tested it locally):

```suggestion
        except requests.exceptions.RequestException as e:
            if e.response is None:
                event.fail(f"unable to connect on port {self._workload_config.port}")
            else:
                event.fail(f"workload responded with code {e.response.status_code}")

```